kernel/arch/dreamcast/hardware/pvr/pvr_dma.c

```diff
@@ -118,21 +126,8 @@ int pvr_dma_transfer(const void *src, uintptr_t dest, size_t count,
         return -1;
     }
 
-    if(DMAC_CHCR2 & 0x1)  /* DE bit set so we must clear it */
-        DMAC_CHCR2 &= ~0x1;
-
-    if(DMAC_CHCR2 & 0x2)  /* TE bit set so we must clear it */
-        DMAC_CHCR2 &= ~0x2;
-
-    DMAC_SAR2 = src_addr;
-    DMAC_DMATCR2 = count / 32;
-    DMAC_CHCR2 = 0x12c1;
-
-    if((DMAC_DMAOR & DMAOR_STATUS_MASK) != DMAOR_NORMAL_OPERATION) {
-        dbglog(DBG_ERROR, "pvr_dma: Failed DMAOR check\n");
-        errno = EIO;
+    if(dma_transfer(&pvr_dma_config, 0, src_addr, count, NULL))
```

Comment:
I guess it is probably better to remove direct access to DMAC here at all and use only the Holly registers for both PVR/TA DMAs as it do for all other DMA transfers on System Bus. And your implementation can be used for those things that the Holly does not allow and for SCIF/SCI or something.

Reply:
I'm just converting the driver. It used DMAC before, it uses DMAC now.

Reply:
> I'm just converting the driver. It used DMAC before, it uses DMAC now.

Yes, I understand, there are no complaints against you. I'm speaking in general.